### DISCUSSION TITLE
Is it safe to use context keyword in production

### DISCUSSION THREAD

**Discussion by yerzhant** [arrow-kt/arrow#3130] [General]
In the [docs](https://arrow-kt.io/learn/typed-errors/working-with-typed-errors/) context receivers are used without a notice that it’s a prototype. Is it safe to use it in production?

**Comment by kyay10** [comment]
This question is more so about Kotlin itself.
To answer it though, context receivers were eventually deprecated, so no, it wasn't "safe to use". Context parameters, right now, are still experimental. However, the compiler itself uses them in production, so they're not as experimental as you may think.
